### ISSUE TITLE
out-of-fold numeric encoding

### ISSUE THREAD

**Issue by muyinanhai** [ffyu/Kaggle-Homesite-Quote-Conversion#1]
Would you mind sharing your encoding solution?

**Comment by ffyu** [comment]
The encoding is used for categorical features. If we want to encode categorical feature "V1", the procedures are summarized as follows:

1) Split the train into 5 folds (fold 1, 2, 3, 4, 5)
2) Use fold 2, 3, 4, 5 to calculate the mean of y for corresponding category of "V1" in fold 1
3) Use fold 1, 3, 4, 5 to calculate the mean of y for corresponding category of "V1" in fold 2
4) Repeat steps 2) and 3) until all folds have their mean for V1 ("V1_mean") calculated.
5) Use the training set to calculate the mean of y for corresponding category of "V1" in test set

Now the additional feature "V1_mean" should be generated for both the training set (via steps 1-4) and the test set (via step 5). It just acts like a normal extra feature. Since this method is using out-of-fold encoding, it should be safe against information leakage.
